index.js

```diff
@@ -66,7 +72,8 @@ module.exports = class Hyperswarm extends EventEmitter {
     this._serverConnections = 0
     this._firewall = firewall
 
-    this.dht.on('network-change', this._handleNetworkChange.bind(this))
+    this._handleNetworkChange = this._handleNetworkChange.bind(this)
+    if (!this._root) this.dht.on('network-change', this._handleNetworkChange)
```

Comment:
we should only setup a single listener on the root and call handleWorkChange on all the sessions from that

Reply:
Isn't doing that? Like time ago you said this and should be good now

Reply:
If `this._root` is defined it means that it's a session, so we doing `if (!this._root)` means that is the actual root so listener is only added on that first root

Reply:
And `_handleNetworkChange` is called here:
https://github.com/holepunchto/hyperswarm/pull/130/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R318

Reply:
ah ok, yes, i think you are right